Basic implementation of metadata-transformer

zomglings opened the pull request:
This is intended to be a minimal functional implementation of the Metadata Transformer that someone would be able to use in production.

Commit:
Integrated with web3 client

zomglings committed:
Added sample.env

zomglings committed:
Added Typescript type definition for ERC721Metadata

zomglings committed:
Working implementation of Inventory transformer

zomglings committed:
Reorganized code to make it a library

zomglings committed:
added .vscode/ to .gitignore

zomglings commented:
To test this, you can run:

```
export METADATA_TRANSFORMER_PORT="6374"
export METADATA_TRANSFORMER_WEB3_PROVIDER_URI="https://polygon-rpc.com"

npm run polygon-inventories
```

Then from a different shell, to test how the Metadata Transformer handles ERC721 tokens which do not have an Inventory:

```
curl localhost:6374/transformed/0xdC0479CC5BbA033B3e7De9F178607150B3AbCe1f/42
```

This should match the metadata returned by:

```
curl localhost:6374/raw/0xdC0479CC5BbA033B3e7De9F178607150B3AbCe1f/42
```

Compare this to:

```
curl localhost:6374/transformed/0xC740674d2DafF5e59284Fc10a39C862A53BF627D/42
```

Which should look very different from:

```
curl localhost:6374/raw/0xC740674d2DafF5e59284Fc10a39C862A53BF627D/42
```

kellan-simiotics reviewed (dismissed):
Looks good.

kompotkot reviewed (dismissed):
tested locally works good

zomglings committed:
Modified "run" function

It now creates the `chain` internally, which means that users can
specify an arbitrary starting point - not just `tokenURI` - and have it
served at `/raw`.

zomglings committed:
removed `tokenURI` import in server.ts

zomglings committed:
Added `fresh` transformer

zomglings committed:
Stopgap error handling

zomglings committed:
Name: `Metadata Transformer`

zomglings committed:
Configured typescript to emit declarations

zomglings committed:
... and declaration source maps

zomglings committed:
Added "prepare" script in package.json

zomglings committed:
Getting things ready for `npm publish`
